pkg/scale/decode_test.go

```diff
@@ -70,6 +71,103 @@ func Test_decodeState_decodeBytes(t *testing.T) {
 	}
 }
 
+var maxLengthDecodeTests = tests{
```

Comment:
make this a function returning a `tests`, so we can be (more) sure there is no funny mutation between each test using it.